hwcomposer/hwcomposer_backend_v20.cpp

```diff
@@ -204,14 +201,18 @@ void HWC2Window::present(HWComposerNativeWindowBuffer *buffer)
 
     QPA_HWC_TIMING_SAMPLE(setTime);
 
-    if (lastPresentFence != -1) {
-        sync_wait(lastPresentFence, -1);
-        close(lastPresentFence);
+    // HWC2 present fences signal when the frame n is displayed on screen
+    // and the buffer for the previous frame n-1 is no longer needed.
+    if (m_previousBuffer != nullptr) {
+        setFenceBufferFd(m_previousBuffer, presentFence);
+        m_previousBuffer->common.decRef(&m_previousBuffer->common);
```

Comment:
don't forget to decRef it in ~HWC2Window(). closing the stored fd would be nice, too ;)